x/interchainstaking/keeper/proposal_handler.go

```diff
@@ -210,7 +227,25 @@ func (k *Keeper) HandleUpdateZoneProposal(ctx sdk.Context, p *types.UpdateZonePr
 				return err
 			}
 			zone.Is_118 = boolValue
+		case "transfer_channel":
+			_, found := k.IBCKeeper.ChannelKeeper.GetChannel(ctx, transfertypes.PortID, change.Value)
+			if !found {
+				return errors.New("unable to fetch channel")
+			}
+			currentChannel, found := k.IBCKeeper.ChannelKeeper.GetChannel(ctx, transfertypes.PortID, zone.TransferChannel)
+			if !found {
+				return errors.New("unable to fetch channel")
+			}
+
+			// check if escrow account has balance of the denom belong to current channel
+			ibcDenom := utils.DeriveIbcDenom("transfer", zone.TransferChannel, "transfer", currentChannel.Counterparty.ChannelId, zone.BaseDenom)
```

Comment:
escrow account on our side won't contain the ibc representation of baseDenom; they will have been minted and sent to the user - therefore we need to check either (or both?):
- the bank module for supply of the above ibcDenom (have any assets been sent _to_ QS from the host zone
- escrow module contains zone.LocalDenom (have any qassets been sent over the current channel)

I suspect the latter is more important?

Reply:
> the bank module for supply of the above ibcDenom (have any assets been sent to QS from the host zone
I think changing the transfer_channel doesn't affect this, so we only need to check for the latter. wdyt?

Reply:
Agree